dts/bindings/memory-controllers/infineon,xmc4xxx-ebu.yaml

```diff
@@ -0,0 +1,216 @@
+
+include: [base.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  "#address-cells":
+    required: true
+    const: 1
+
+  "#size-cells":
+    required: true
+    const: 0
+
+  clk-config:
+    type: int
+    required: true
+    description: |
+      Configures Clock Control (CLC) register. Controls which clock is used for the EBU and
+      the clock divider. Use XMC4XXX_EBU_SET_CLC(sync, div2, ebudiv).
+
+  modes-config:
+    type: int
+    required: true
+    description: |
+      Sets Modes Configuration (MODCON) register. This mainly sets the arbitration parameters.
+      Also it is used to configure whether address latch enable (ALE) is active high or low and whether SDRAM
+      pins CKE, SDCLKO, CAS, and RAS are in tri-state when the EBU does not own the external bus.
+      Use macro XMC4XXX_EBU_SET_MODCON(sdtri, extlock, arbsync, arbmode, timeoutc, ale).
+
+  gpio-control-config:
+    type: int
+    required: true
+    description: |
+      Configures USERCON register. This register allows the user to free unused gpio pins.
+      Use macro XMC4XXX_EBU_SET_USERCON(addio, advio).
+
+  sdram-control:
+    type: int
+    required: false
+    description: |
+      Configures SDRMCON register.
+      Use macro XMC4XXX_EBU_SET_SDRMCON(sdcmsel, pwr_mode, clkdis, crce, bankm,
+      rowm, crc, crcd, awidth, crp, crsc, crfsh, cras).
+
+  sdram-operation-mode:
+    type: int
+    required: false
+    description: |
+      Configures SDRMOD register.
+      Use macro XMC4XXX_EBU_SDRMOD(xba, xopm, coldstart, opmode, caslat, burstl).
+
+  sdram-refresh-control:
+    type: int
+    required: false
+    description: |
+      Configures SDRMREF register.
+      Use macro XMC4XXX_EBU_SDRMREF(res_dly, arfsh, selfrex_dly, erfshc, autoselfr,
+      selfrex, refreshr, refreshc).
+
+child-binding:
+  description: |
+    Child device nodes are representing devices connected to the EBU bus.
+
+  properties:
+    reg:
+      type: int
+      required: true
+      description: Selects the region to configure. All the following properties apply to this region.
+
+    device-type:
+      type: string
+      required: true
+      description: Device type.
+      enum:
+        - "Muxed Asynchronous Type"
+        - "Muxed Burst Type"
+        - "NAND Flash"
+        - "Muxed Cellular RAM"
+        - "Demuxed Asynchronous Type"
+        - "Demuxed Burst Type"
+        - "Demuxed Page Mode"
+        - "Demuxed Cellular RAM"
+        - "SDRAM"
+
+    address-bus-width:
+      type: string
+      required: true
+      description: Address bus width configuration.
+      enum:
+        - "16-bit"
+        - "Twin-16-bit"
+        - "32-bit"
+
+    region-enable:
+      type: boolean
+      description: |
+        Enables the address space 0x6x000000-. This address space supports code
+        execution.
+
+    alternate-region-enable:
+      type: boolean
+      description: Enables the non-executable alternate address space 0xax000000-.
```

Comment:
This description doesn't seem finished.

Reply:
I added more details in the description.